Greenheart/swipeforfuture.com
# Feature/refactor rewrite game scenario

This branch also includes the changes made for `refactor: replace GameScenario with a more generic Game type` and also adds a feature flag which enables use of a reimplementation of the BasicGameScenario using BasicGame. This PR will:
* Add BasicGame as a new extensible Game\<P\> concept
* Add a drop in replacement for BasicGameScenario
* Make WorldStateExtensions a legacy format in favour of StateExtensions with a backwards compatibility layer

This PR should not cause any user facing changes.

To test:
* Build and run `npm start`
* Fixed with latest content update.~Make sure you are using latest data with `modifiers` and not `modifier` (currently the target online data is a mismatch)~
* Verify that the game behaves as expected
